src/opm/output/data/InterRegFlowMap.cpp

```diff
@@ -0,0 +1,588 @@
+Opm::data::InterRegFlowMap::CSR::getWindow(const int i, const int j) const
+{
+    if ((i < 0) || (i >= static_cast<int>(this->numRows())) ||
+        (j < 0) || (j >  this->maxColIdx()))
+    {
+        // Entity pair IDs out of range.
+        return std::nullopt;
+    }
+
+    auto begin = this->ja().begin() + this->ia()[i + 0];
+    auto end   = this->ja().begin() + this->ia()[i + 1];
+
+    auto pos = std::lower_bound(begin, end, j);
+    if ((pos == end) || (*pos > j)) {
+        // Entity 'j' does not connect to entity 'i'.
+        return std::nullopt;
+    }
+
+    // Entity 'j' connects to 'i'.  Form read-only view into sub-range
+    // pertaining to this entity pair.
+    const auto sz = ReadOnlyWindow::bufferSize();
+    const auto windowID = pos - this->ja().begin();
+    auto start = this->sa().begin() + windowID*sz;
+
+    return { ReadOnlyWindow{ start, start + sz } };
+}
+
+Opm::data::InterRegFlowMap::Offset
+Opm::data::InterRegFlowMap::CSR::numRows() const
+{
+    return this->ia().empty()
+        ? 0 : this->ia().size() - 1;
+}
+
+int Opm::data::InterRegFlowMap::CSR::maxRowIdx() const
+{
+    return this->numRows_ - 1;
+}
+
+int Opm::data::InterRegFlowMap::CSR::maxColIdx() const
+{
+    return this->numCols_ - 1;
+}
+
+const Opm::data::InterRegFlowMap::Start&
+Opm::data::InterRegFlowMap::CSR::ia() const
+{
+    return this->ia_;
+}
+
+const Opm::data::InterRegFlowMap::Neighbours&
+Opm::data::InterRegFlowMap::CSR::ja() const
+{
+    return this->ja_;
+}
+
+const Opm::data::InterRegFlowMap::RateBuffer&
+Opm::data::InterRegFlowMap::CSR::sa() const
+{
+    return this->sa_;
+}
+
+std::vector<int>
+Opm::data::InterRegFlowMap::CSR::expandStartPointers() const
+{
+    auto rowIdx = std::vector<int>{};
+
+    if (this->ia_.empty()) {
+        return rowIdx;
+    }
+
+    rowIdx.reserve(this->ia_.back());
+
+    auto row = 0;
+
+    const auto m = this->ia_.size() - 1;
+    for (auto i = 0*m; i < m; ++i, ++row) {
+        const auto n = this->ia_[i + 1] - this->ia_[i + 0];
+
+        rowIdx.insert(rowIdx.end(), n, row);
+    }
+
+    return rowIdx;
+}
+
+void Opm::data::InterRegFlowMap::CSR::clear()
+{
+    this->ia_.clear();
+    this->ja_.clear();
+    this->sa_.clear();
+    this->compressedIdx_.clear();
+
+    this->numRows_ = 0;
+    this->numCols_ = 0;
+}
+
+void
+Opm::data::InterRegFlowMap::
+CSR::assemble(const Neighbours& rows,
+              const Neighbours& cols,
+              const int         maxRowIdx,
+              const int         maxColIdx)
+{
+    auto i = this->expandStartPointers();
+    i.insert(i.end(), rows.begin(), rows.end());
+
+    auto j = this->ja_;
+    j.insert(j.end(), cols.begin(), cols.end());
+
+    const auto thisNumRows = std::max(this->numRows_, maxRowIdx + 1);
+    const auto thisNumCols = std::max(this->numCols_, maxColIdx + 1);
+
+    this->accumulateRowEntries(thisNumRows, i);
+
+    this->createGraph(i, j);
+
+    this->numRows_ = thisNumRows;
+    this->numCols_ = thisNumCols;
+}
```

Comment:
This is true.  Doing it this way enables merging other CSR contributions into an existing CSR structure, albeit by going through a coordinate format first.  If we were to merge the two more directly it would lead to a lot more data movement.